src/browser/preload_scripts.ts

```diff
@@ -88,24 +89,77 @@ function loadScript(identity: Identity, preloadScript: PreloadScript): Promise<P
         };
 
         log(`Started loading preload script for URL [${url}]`);
-        Window.setWindowPreloadState(identity, {...preloadScript, state: 'load-started'});
+        setWindowPreloadState(identity, {...preloadScript, state: 'load-started'});
 
         if (!scriptPath) {
             log(`Failed loading preload script for URL [${url}]: preload script wasn't downloaded`);
-            Window.setWindowPreloadState(identity, {...preloadScript, state: 'load-failed'});
+            setWindowPreloadState(identity, {...preloadScript, state: 'load-failed'});
             return resolve({...preloadScript, _content: ''});
         }
 
         readFile(scriptPath, 'utf8', (readError: Error, data: string) => {
             if (readError) {
                 log(`Failed loading preload script for URL [${url}] from path [${scriptPath}]: ${readError}`);
-                Window.setWindowPreloadState(identity, {...preloadScript, state: 'load-failed'});
+                setWindowPreloadState(identity, {...preloadScript, state: 'load-failed'});
                 resolve({...preloadScript, _content: ''});
             } else {
                 log(`Succeeded loading preload script for URL [${url}] from path [${scriptPath}]`);
-                Window.setWindowPreloadState(identity, {...preloadScript, state: 'load-succeeded'});
+                setWindowPreloadState(identity, {...preloadScript, state: 'load-succeeded'});
                 resolve({...preloadScript, _content: data});
             }
         });
     });
 }
+export function setWindowPreloadState (identity: Identity, payload: { state: any; mandatory?: boolean; url: any; allDone?: any; }) {
+    const { uuid, name } = identity;
+    const { url, state, allDone } = payload;
+    const updateTopic = allDone ? 'preload-scripts-state-changed' : 'preload-scripts-state-changing';
+    const frameInfo = getInfoByUuidFrame(identity);
+    let ofEntity: InjectableContext | false;
+    if (frameInfo.entityType === 'iframe') {
+        ofEntity = getEntityByUuidFrame(frameInfo.parent.uuid, frameInfo.parent.name);
+    } else {
+        ofEntity = getEntityByUuidFrame(uuid, name);
+    }
+
+    if (!ofEntity) {
+        return writeToLog('info', `setWindowPreloadState missing openfinWindow ${uuid} ${name}`);
```

Comment:
small nitpick, should this say OpenFinWindow or View?

Reply:
Ah, nvm, it's only for Windows.

Reply:
Also should we update the name of the function?

Reply:
Trying to keep this code close to the original for now.